Automattic/wp-calypso
# Checkout: Copy tax country code into hidden Ebanx country field

#### Background

Currently, our credit card form uses Stripe unless the user's currency is in BRL _and_ the user selects "Brazil" as their country in the tax location / contact info step of checkout, in which case we display and use Ebanx for processing. The Ebanx card form has many more fields than the Stripe form, most of them specific to Brazil.

The component that displays these form fields (`CountrySpecificPaymentFields`) decides what fields to display based on the `countryCode` provided to it. However, that `countryCode` comes from a contact details state that is separate from the contact details state used for the rest of checkout (the rest of checkout is concerned with tax address and/or domain registration address), and since `CountrySpecificPaymentFields` does not display its own country selector (which would be redundant for most customers), we need to provide it programmatically.

The server-side validation of the Ebanx form fields also requires a country code to be included.

In order to always display the correct fields and pass validation, when checkout was redesigned, we hard-coded the country code for the Ebanx form to `BR`. 

#### The Problem

Recently we decided we'd like to be able to display the Ebanx form for customers with BRL currency who chose a country other than "Brazil" in the tax location / contact info step of checkout (see 962-gh-Automattic/payments-shilling). This will be enabled by D83605-code. If we do that, however, the hard-coded country code in the Ebanx form may no longer be correct for everyone.

#### Changes in this PR

Here we remove the hard-coded country code in the Ebanx form and replace it with the country code selected in the tax location / contact info step.

#### Screenshots

The Stripe card form (not affected by this change):
<img width="491" alt="Screen Shot 2022-09-01 at 6 57 22 PM" src="https://user-images.githubusercontent.com/2036909/188026330-b63cf72c-4348-4c96-82a7-54e1bcb1ae97.png">

The Ebanx card form when "Brazil" is _not selected_:
<img width="494" alt="Screen Shot 2022-09-01 at 6 53 20 PM" src="https://user-images.githubusercontent.com/2036909/188025967-30bebc4e-6b3f-40ab-8fd1-f78f0c35d31f.png">

The Ebanx card form when "Brazil" is selected:
<img width="486" alt="Screen Shot 2022-09-01 at 6 53 45 PM" src="https://user-images.githubusercontent.com/2036909/188025977-aaf9050d-875f-45b4-b465-9e5878d04a1c.png">


#### Testing Instructions

- With D83605-code applied to your sandbox, sandbox the API.
- Using Store Admin, set your currency to BRL.
- Add a product to your cart and visit checkout.
- Fill out the checkout form but make sure to set your country to something other than "Brazil".
- At the payment method step, select "Credit or debit card".
- Verify that you see the Ebanx form rather than the Stripe form (see screenshots above).
- Fill out the form and verify that the validation and purchase works correctly.
